tests/kernel/pthread/src/pthread.c

```diff
@@ -26,18 +28,6 @@ PTHREAD_BARRIER_DEFINE(barrier, N_THR);
 
 K_SEM_DEFINE(main_sem, 0, 2*N_THR);
 
-/* Should be exactly N_THR threads defined with the first argument
- * being a zero-indexed sequential ID
- */
-K_THREAD_DEFINE(thread0, STACKSZ, thread_top, (void *)0, NULL, NULL,
-		K_HIGHEST_THREAD_PRIO, 0, K_NO_WAIT);
-
-K_THREAD_DEFINE(thread1, STACKSZ, thread_top, (void *)1, NULL, NULL,
-		K_HIGHEST_THREAD_PRIO, 0, K_NO_WAIT);
-
-K_THREAD_DEFINE(thread2, STACKSZ, thread_top, (void *)2, NULL, NULL,
-		K_HIGHEST_THREAD_PRIO, 0, K_NO_WAIT);
-
```

Comment:
I don't have any particular complaints with any of these changes.  But it might be worth adding them as a separate test.  Using pthread synchronization with Zephyr-native threads is a supported use case, and this removes testing for that while it adds testing for POSIX-native thread APIs.  Why not both?

Reply:
Idle here was to convert the pthread application entirely POSIX compliant and avoid using Zephyr APIs.
Is your suggestion to keep this application untouched and add posix compliant version of this application (pthread) separately ?